src/main/java/database/service/service/JsonDatabaseService.java

```diff
@@ -0,0 +1,195 @@
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class JsonDatabaseService implements DatabaseService {
+    private final Settings settings = new Settings();
+    private final ObjectMapper objectMapper = new ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);
+    private Map<String, Integer> entityIds = new HashMap<>();
+
+    public JsonDatabaseService() throws CriticalDatabaseException {
+    }
+
+    @Override
+    public boolean createTable(Class<? extends BaseEntity> entityClass) throws IOException {
+        File jsonDatabaseFile = new File(getDatabasePath(entityClass));
+        jsonDatabaseFile.createNewFile();
+        Files.writeString(jsonDatabaseFile.toPath(), "[]");
+        return Files.exists(jsonDatabaseFile.toPath());
+    }
+
+    @Override
+    public boolean deleteTable(Class<? extends BaseEntity> entityClass) throws IOException {
+        Path databasePath = Path.of(getDatabasePath(entityClass));
+        Files.delete(databasePath);
+        return !Files.exists(databasePath);
+    }
+
+    @Override
+    public <T extends BaseEntity> T addNewRecordToTable(T entity) throws DatabaseDoesNotExistException, IOException {
+        Class<? extends BaseEntity> entityClass = entity.getClass();
+        Path databasePath = Path.of(getDatabasePath(entityClass));
+
+        if (!Files.exists(databasePath)) {
+            throw new DatabaseDoesNotExistException("Database does not exist.");
+        }
+
+        List<T> entities = deserializeEntities(entityClass, Files.readString(databasePath));
+        String entityClassName = entityClass.getName();
+        var lastUsedId = entityIds.get(entityClassName);
+
+        if (lastUsedId == null) {
+            if (entities.isEmpty()) {
+                entityIds.put(entityClassName, -1);
+            } else {
+                int lastEntityId = entities.get(entities.size() - 1).getId();
+                entityIds.put(entityClassName, lastEntityId);
+            }
+        }
+
+        entity.setId(entityIds.merge(entityClassName, 1, Integer::sum));
+        entities.add(entity);
+
+        objectMapper.writeValue(databasePath.toFile(), entities);
+        return entity;
+    }
+
+    @Override
+    public <T extends BaseEntity, I> T updateRecordInTable(T entity, I id) throws IOException,
+            ProvidedIdDoesNotExistException {
+        Class<? extends BaseEntity> entityClass = entity.getClass();
+        Path databasePath = Path.of(getDatabasePath(entityClass));
+
+        List<T> entities = deserializeEntities(entityClass, Files.readString(databasePath));
+
+        Optional<T> optionalEntity = entities.stream().filter(e -> id.equals(e.getId())).findFirst();
```

Comment:
You can use `.orElseThrow(() -> new ProvidedIdDoesNotExistException("Entity with provided Id does not exist."))` to avoid of `if-else` block.